--- thirdparty/LaBraM/engine_for_finetuning.py	2026-01-19 14:47:38.540255299 +0900
+++ finetune/LaBraM/engine_for_finetuning.py	2026-04-06 21:07:10.077052382 +0900
@@ -75,3 +75,3 @@
         else:
-            with torch.cuda.amp.autocast():
+            with torch.amp.autocast('cuda'):
                 loss, output = train_class_batch(
@@ -179,3 +179,3 @@
         # compute output
-        with torch.cuda.amp.autocast():
+        with torch.amp.autocast('cuda'):
             output = model(EEG, input_chans=input_chans)
@@ -189,3 +189,3 @@
 
-        results = utils.get_metrics(output.numpy(), target.numpy(), metrics, is_binary)
+        # results = utils.get_metrics(output.numpy(), target.numpy(), metrics, is_binary)
         pred.append(output)
@@ -195,4 +195,4 @@
         metric_logger.update(loss=loss.item())
-        for key, value in results.items():
-            metric_logger.meters[key].update(value, n=batch_size)
+        # for key, value in results.items():
+        #     metric_logger.meters[key].update(value, n=batch_size)
         #metric_logger.meters['acc5'].update(acc5.item(), n=batch_size)
