--- thirdparty/LaBraM/run_class_finetuning.py	2026-04-06 00:53:40.086809828 +0900
+++ finetune/LaBraM/run_class_finetuning.py	2026-04-11 21:24:58.045280139 +0900
@@ -214,3 +214,3 @@
     if args.dataset == 'TUAB':
-        train_dataset, test_dataset, val_dataset = utils.prepare_TUAB_dataset("path/to/TUAB")
+        train_dataset, test_dataset, val_dataset = utils.prepare_TUAB_dataset("../../datasets/tuab/labram")
         ch_names = ['EEG FP1', 'EEG FP2-REF', 'EEG F3-REF', 'EEG F4-REF', 'EEG C3-REF', 'EEG C4-REF', 'EEG P3-REF', 'EEG P4-REF', 'EEG O1-REF', 'EEG O2-REF', 'EEG F7-REF', \
@@ -220,4 +220,12 @@
         metrics = ["pr_auc", "roc_auc", "accuracy", "balanced_accuracy"]
+    elif args.dataset == 'WORKLOAD':
+        train_dataset, test_dataset, val_dataset = get_EEGMAT_EAREEG_ISRUC_dataset('WORKLOAD')
+        ch_names = ['EEG FP1', 'EEG FP2-REF', 'EEG F3-REF', 'EEG F4-REF', 'EEG C3-REF', 'EEG C4-REF', 'EEG P3-REF', 'EEG P4-REF', 'EEG O1-REF', 'EEG O2-REF', 'EEG F7-REF', \
+                    'EEG F8-REF', 'EEG T3-REF', 'EEG T4-REF', 'EEG T5-REF', 'EEG T6-REF', 'EEG A1-REF', 'EEG A2-REF', 'EEG FZ-REF', 'EEG CZ-REF', 'EEG PZ-REF', 'EEG T1-REF', 'EEG T2-REF']
+        ch_names = [name.split(' ')[-1].split('-')[0] for name in ch_names]
+        ch_names = ch_names[:16]
+        args.nb_classes = 1
+        metrics = ["pr_auc", "roc_auc", "accuracy", "balanced_accuracy"]
     elif args.dataset == 'TUEV':
-        train_dataset, test_dataset, val_dataset = utils.prepare_TUEV_dataset("path/to/TUEV")
+        train_dataset, test_dataset, val_dataset = utils.prepare_TUEV_dataset("../../datasets/tuev/labram")
         ch_names = ['EEG FP1-REF', 'EEG FP2-REF', 'EEG F3-REF', 'EEG F4-REF', 'EEG C3-REF', 'EEG C4-REF', 'EEG P3-REF', 'EEG P4-REF', 'EEG O1-REF', 'EEG O2-REF', 'EEG F7-REF', \
@@ -227,4 +235,31 @@
         metrics = ["accuracy", "balanced_accuracy", "cohen_kappa", "f1_weighted"]
+    elif args.dataset == 'EarEEG':
+        train_dataset, test_dataset, val_dataset = get_EEGMAT_EAREEG_ISRUC_dataset('EarEEG')
+        ch_names = ['EEG FP1', 'EEG FP2-REF', 'EEG F3-REF', 'EEG F4-REF', 'EEG C3-REF', 'EEG C4-REF', 'EEG P3-REF', 'EEG P4-REF', 'EEG O1-REF', 'EEG O2-REF', 'EEG F7-REF', \
+                    'EEG F8-REF', 'EEG T3-REF', 'EEG T4-REF', 'EEG T5-REF', 'EEG T6-REF', 'EEG A1-REF', 'EEG A2-REF', 'EEG FZ-REF', 'EEG CZ-REF', 'EEG PZ-REF', 'EEG T1-REF', 'EEG T2-REF']
+        ch_names = [name.split(' ')[-1].split('-')[0] for name in ch_names]
+        ch_names = ch_names[:4]
+        args.nb_classes = 6
+        metrics = ["accuracy", "balanced_accuracy", "cohen_kappa", "f1_weighted"]
+    elif args.dataset == 'ISRUC':
+        train_dataset, test_dataset, val_dataset = get_EEGMAT_EAREEG_ISRUC_dataset('ISRUC')
+        ch_names = ['EEG FP1', 'EEG FP2-REF', 'EEG F3-REF', 'EEG F4-REF', 'EEG C3-REF', 'EEG C4-REF', 'EEG P3-REF', 'EEG P4-REF', 'EEG O1-REF', 'EEG O2-REF', 'EEG F7-REF', \
+                    'EEG F8-REF', 'EEG T3-REF', 'EEG T4-REF', 'EEG T5-REF', 'EEG T6-REF', 'EEG A1-REF', 'EEG A2-REF', 'EEG FZ-REF', 'EEG CZ-REF', 'EEG PZ-REF', 'EEG T1-REF', 'EEG T2-REF']
+        ch_names = [name.split(' ')[-1].split('-')[0] for name in ch_names]
+        args.nb_classes = 5
+        ch_names = ch_names[:6]
+        metrics = ["accuracy", "balanced_accuracy", "cohen_kappa", "f1_weighted"]
+
+    # We use the leading x channels as ch_names for new datasets
+    # Contents of ch_names is not important here since we will reset the channel_embedding later
+
     return train_dataset, test_dataset, val_dataset, ch_names, metrics
 
+def get_EEGMAT_EAREEG_ISRUC_dataset(dataset):
+    from data_loaders import get_dataloaders
+    train_set = get_dataloaders(dataset, 'train', 200)
+    test_set = get_dataloaders(dataset, 'test', 200)
+    val_set = get_dataloaders(dataset, 'val', 200)
+
+    return train_set, test_set, val_set
 
@@ -338,3 +373,3 @@
         else:
-            checkpoint = torch.load(args.finetune, map_location='cpu')
+            checkpoint = torch.load(args.finetune, weights_only=False, map_location='cpu')
 
@@ -372,2 +407,22 @@
 
+    # For EarEEG and ISRUC, since the length of the signal is longer than 16, we need to patch that to at least 30
+    # And for all EarEEG ISRUC Workload, channel names are not match, so we need to reset pos_embed
+    if args.dataset in ['EarEEG', 'ISRUC']:
+        time_embed = model.time_embed # We always expand the time_embed to 60 seconds even though it may not been used
+        time_shape = list(time_embed.shape)
+        if time_shape[-2] < 60:
+            time_shape[-2] = 60 - time_shape[-2]
+            time_pending = torch.empty(time_shape, dtype=time_embed.dtype, device=time_embed.device)
+            torch.nn.init.trunc_normal_(time_pending, std=0.02)
+
+            time_embed = torch.concat([time_embed, time_pending], dim = -2)
+            time_embed.requires_grad_()
+            model.time_embed = torch.nn.Parameter(time_embed, True)
+
+    if args.dataset in ['WORKLOAD', 'EarEEG', 'ISRUC']:
+        pos_embed = model.pos_embed
+        if pos_embed is not None:
+            torch.nn.init.trunc_normal_(pos_embed, std=0.02) # we don't reset [cls] token
+            # Since it has been reset, the ch name will be not important
+
     model.to(device)
