--- thirdparty/LaBraM/run_class_finetuning.py	2026-04-06 00:53:40.086809828 +0900
+++ finetune/LaBraM/run_class_finetuning.py	2026-04-06 22:30:38.469111295 +0900
@@ -214,3 +214,3 @@
     if args.dataset == 'TUAB':
-        train_dataset, test_dataset, val_dataset = utils.prepare_TUAB_dataset("path/to/TUAB")
+        train_dataset, test_dataset, val_dataset = utils.prepare_TUAB_dataset("../../datasets/tuab/labram")
         ch_names = ['EEG FP1', 'EEG FP2-REF', 'EEG F3-REF', 'EEG F4-REF', 'EEG C3-REF', 'EEG C4-REF', 'EEG P3-REF', 'EEG P4-REF', 'EEG O1-REF', 'EEG O2-REF', 'EEG F7-REF', \
@@ -221,3 +221,3 @@
     elif args.dataset == 'TUEV':
-        train_dataset, test_dataset, val_dataset = utils.prepare_TUEV_dataset("path/to/TUEV")
+        train_dataset, test_dataset, val_dataset = utils.prepare_TUEV_dataset("../../datasets/tuev/labram")
         ch_names = ['EEG FP1-REF', 'EEG FP2-REF', 'EEG F3-REF', 'EEG F4-REF', 'EEG C3-REF', 'EEG C4-REF', 'EEG P3-REF', 'EEG P4-REF', 'EEG O1-REF', 'EEG O2-REF', 'EEG F7-REF', \
@@ -338,3 +338,3 @@
         else:
-            checkpoint = torch.load(args.finetune, map_location='cpu')
+            checkpoint = torch.load(args.finetune, weights_only=False, map_location='cpu')
 
