--- thirdparty/EEGPT/downstream_tueg/Modules/models/EEGPT_mcae_finetune_change_tuev.py	2026-05-10 22:43:54.119493116 +0900
+++ finetune/EEGPT/downstream_tueg/Modules/models/EEGPT_mcae_finetune_change_tuev.py	2026-05-10 23:49:02.494313995 +0900
@@ -569,3 +569,3 @@
         super(Conv1dWithConstraint, self).__init__(*args, **kwargs)
-    @autocast(True)
+    # @autocast(True)
     def forward(self, x):
@@ -583,3 +583,3 @@
         super(LinearWithConstraint, self).__init__(*args, **kwargs)
-    @autocast(True)
+    # @autocast(True)
     def forward(self, x):
@@ -601,3 +601,3 @@
 
-    @autocast(True)
+    # @autocast(True)
     def forward(self, x):
