--- thirdparty/EEGPT/downstream/finetune_EEGPT_SleepEDF.py	2026-01-13 22:30:32.507362243 +0900
+++ finetune/EEGPT/downstream/finetune_EEGPT_EarEEG.py	2026-05-11 09:54:19.455014074 +0900
@@ -38,4 +38,4 @@
         super().__init__()    
-        self.chans_num = len(use_channels_names)
-        self.num_class = 5
+        self.chans_num = 4 #len(use_channels_names)
+        self.num_class = 6
         # init model
@@ -58,6 +58,6 @@
         self.target_encoder = target_encoder
-        self.chans_id       = target_encoder.prepare_chan_ids(use_channels_names)
+        self.chans_id       = None # target_encoder.prepare_chan_ids(use_channels_names)
         
         # -- load checkpoint
-        pretrain_ckpt = torch.load(load_path)
+        pretrain_ckpt = torch.load(load_path, weights_only=False)
         
@@ -70,4 +70,5 @@
         self.target_encoder.load_state_dict(target_encoder_stat)
+        self.target_encoder.chan_embed.reset_parameters()
 
-        self.chan_conv       = Conv1dWithConstraint(2, self.chans_num, 1, max_norm=1)
+        self.chan_conv       = Conv1dWithConstraint(self.chans_num, self.chans_num, 1, max_norm=1)
         
@@ -92,3 +93,3 @@
         self.target_encoder.eval()
-        z = self.target_encoder(x, self.chans_id.to(x))
+        z = self.target_encoder(x, self.chans_id)
         
@@ -194,11 +195,15 @@
 # Train Data Num : 5Class: 36914 13604 34722 7922 15398
-subjects = [0, 2, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 23, 24, 25, 26, 29, 30, 31, 32, 33, 34, 35, 37, 38, 40, 42, 44, 45, 46, 47, 48, 49, 51, 52, 53, 54, 55, 56, 57, 58, 59, 61, 62, 63, 64, 65, 66, 71, 72, 73, 74, 75, 76, 77, 81, 82]
-N = len(subjects)//10
-set_all = set(subjects)
-for fold in range(10):
-    set_valid = set(subjects[fold*N:(fold+1)*N])
-    set_train = set_all - set_valid
-
-    train_dataset = torchvision.datasets.DatasetFolder(root="../datasets/downstream/sleep_edf/TrainFold", loader=lambda x: torch.load(x),  extensions=[f'.s{i}' for i in set_train])
-    valid_dataset = torchvision.datasets.DatasetFolder(root="../datasets/downstream/sleep_edf/TrainFold", loader=lambda x: torch.load(x), extensions=[f'.s{i}' for i in set_valid])
+# subjects = [0, 2, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 23, 24, 25, 26, 29, 30, 31, 32, 33, 34, 35, 37, 38, 40, 42, 44, 45, 46, 47, 48, 49, 51, 52, 53, 54, 55, 56, 57, 58, 59, 61, 62, 63, 64, 65, 66, 71, 72, 73, 74, 75, 76, 77, 81, 82]
+# N = len(subjects)//10
+# set_all = set(subjects)
+# for fold in range(10):
+if __name__ == '__main__':
+    # set_valid = set(subjects[fold*N:(fold+1)*N])
+    # set_train = set_all - set_valid
+
+    # train_dataset = torchvision.datasets.DatasetFolder(root="../datasets/downstream/sleep_edf/TrainFold", loader=lambda x: torch.load(x),  extensions=[f'.s{i}' for i in set_train])
+    # valid_dataset = torchvision.datasets.DatasetFolder(root="../datasets/downstream/sleep_edf/TrainFold", loader=lambda x: torch.load(x), extensions=[f'.s{i}' for i in set_valid])
+    import data_loaders
+    train_dataset = data_loaders.get_dataloaders('EarEEG', 'train', 200)
+    valid_dataset = data_loaders.get_dataloaders('EarEEG', 'val', 200)
 
@@ -227,3 +232,6 @@
     lr_monitor = pl.callbacks.LearningRateMonitor(logging_interval='epoch')
-    callbacks = [lr_monitor]
+    cp_callback = pl.callbacks.ModelCheckpoint(
+        './checkpoint/' + 'EEGPT_EarEEG', monitor='valid_cohen_kappa', save_last=True, save_top_k=1, mode='max'
+    )
+    callbacks = [lr_monitor, cp_callback]
 
@@ -233,4 +241,4 @@
                         callbacks=callbacks,
-                        logger=[pl_loggers.TensorBoardLogger('./logs/', name="EEGPT_SLEEPEDF_tb", version=f"fold{fold+1}"), 
-                                pl_loggers.CSVLogger('./logs/', name="EEGPT_SLEEPEDF_csv")])
+                        logger=[pl_loggers.TensorBoardLogger('./logs/', name="EEGPT_EarEEG_tb", version=f"fold{1}"), 
+                                pl_loggers.CSVLogger('./logs/', name="EEGPT_EarEEG_csv")])
 
