--- thirdparty/EEGPT/downstream_tueg/finetune_TUAB_EEGPT.sh	2026-01-13 22:30:32.510962244 +0900
+++ finetune/EEGPT/downstream_tueg/finetune_TUAB_EEGPT.sh	2026-05-10 23:49:02.490713995 +0900
@@ -15,4 +15,5 @@
 # Please refer to `run_mae_pretraining.py` for the meaning of the following hyperreferences
-CUDA_VISIBLE_DEVICES=4,5 OMP_NUM_THREADS=1 python -m torch.distributed.run --nproc_per_node=${GPUS_PER_NODE} \
-        --master_port ${MASTER_PORT} --nnodes=${N_NODES} --node_rank=0 --master_addr="localhost" \
+# CUDA_VISIBLE_DEVICES=4,5 OMP_NUM_THREADS=1 python -m torch.distributed.run --nproc_per_node=${GPUS_PER_NODE} \
+#         --master_port ${MASTER_PORT} --nnodes=${N_NODES} --node_rank=0 --master_addr="localhost" \
+python \
         run_class_finetuning_EEGPT_change.py \
@@ -23,3 +24,3 @@
         --weight_decay 0.05 \
-        --batch_size 100\
+        --batch_size 256\
         --lr 5e-4 \
