--- thirdparty/EEGPT/downstream_tueg/finetune_TUEV_EEGPT.sh
+++ finetune/EEGPT/downstream_tueg/finetune_TUEV_EEGPT.sh
@@ -15,4 +15,5 @@
 # Please refer to `run_class_finetuning_EEGPT_change_tuev.py` for the meaning of the following hyperreferences
-CUDA_VISIBLE_DEVICES=4,5 OMP_NUM_THREADS=1 python -m torch.distributed.run --nproc_per_node=${GPUS_PER_NODE} \
-        --master_port ${MASTER_PORT} --nnodes=${N_NODES} --node_rank=0 --master_addr="localhost" \
+# CUDA_VISIBLE_DEVICES=4,5 OMP_NUM_THREADS=1 python -m torch.distributed.run --nproc_per_node=${GPUS_PER_NODE} \
+#         --master_port ${MASTER_PORT} --nnodes=${N_NODES} --node_rank=0 --master_addr="localhost" \
+python \
         run_class_finetuning_EEGPT_change_tuev.py \
@@ -23,3 +24,3 @@
         --weight_decay 0.05 \
-        --batch_size 400\
+        --batch_size 256\
         --lr 5e-4 \
@@ -27,3 +28,3 @@
         --warmup_epochs 5 \
-        --epochs 30 \
+        --epochs 50 \
         --layer_decay 0.65 \
@@ -35,3 +36,2 @@
         --dataset TUEV\
-        --enable_deepspeed \
         --seed 0
\ No newline at end of file
