--- thirdparty/BIOT/run_binary_supervised.py	2026-01-08 15:46:45.832706321 +0900
+++ finetune/BIOT/run_binary_supervised.py	2026-04-12 06:18:26.470716982 +0900
@@ -26,3 +26,2 @@
 
-
 class LitModel_finetune(pl.LightningModule):
@@ -124,3 +123,12 @@
         return [optimizer]  # , [scheduler]
+    
+    def on_save_checkpoint(self, checkpoint):
+        super().on_save_checkpoint(checkpoint)
+
+        checkpoint['threshold'] = float(self.threshold)
+
+    def on_load_checkpoint(self, checkpoint):
+        super().on_load_checkpoint(checkpoint)
 
+        self.threshold = checkpoint.get('threshold', self.threshold)
 
@@ -134,3 +142,3 @@
 
-    root = "/srv/local/data/TUH/tuh3/tuh_eeg_abnormal/v3.0.0/edf/processed"
+    root = "../../datasets/tuab/biot/"
 
@@ -171,2 +179,43 @@
 
+def prepare_EEGMAT_dataloader(args):
+    # set random seed
+    seed = 12345
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    np.random.seed(seed)
+
+    from data_loaders import get_dataloaders
+
+    train_set = get_dataloaders('WORKLOAD', 'train', 200)
+    val_set = get_dataloaders('WORKLOAD', 'val', 200)
+    test_set = get_dataloaders('WORKLOAD', 'test', 200)
+
+    # prepare training and test data loader
+    train_loader = torch.utils.data.DataLoader(
+        train_set,
+        batch_size=args.batch_size,
+        shuffle=True,
+        drop_last=True,
+        num_workers=args.num_workers,
+        persistent_workers=args.num_workers > 0,
+    )
+    test_loader = torch.utils.data.DataLoader(
+        # TUABLoader(os.path.join(root, "test"), test_files, args.sampling_rate),
+        test_set,
+        batch_size=args.batch_size,
+        shuffle=False,
+        num_workers=args.num_workers,
+        persistent_workers=args.num_workers > 0,
+    )
+    val_loader = torch.utils.data.DataLoader(
+        # TUABLoader(os.path.join(root, "val"), val_files, args.sampling_rate),
+        val_set,
+        batch_size=args.batch_size,
+        shuffle=False,
+        num_workers=args.num_workers,
+        persistent_workers=args.num_workers > 0,
+    )
+    print(len(train_loader), len(val_loader), len(test_loader))
+    return train_loader, test_loader, val_loader
 
@@ -265,2 +314,4 @@
         train_loader, test_loader, val_loader = prepare_TUAB_dataloader(args)
+    elif args.dataset in ['EEGMAT', 'WORKLOAD']:
+        train_loader, test_loader, val_loader = prepare_EEGMAT_dataloader(args)
 
@@ -335,2 +386,4 @@
 
+        if args.dataset in ['EEGMAT', 'WORKLOAD']:
+            model.biot.channel_tokens.reset_parameters()
     else:
@@ -347,3 +400,7 @@
     early_stop_callback = EarlyStopping(
-        monitor="val_auroc", patience=5, verbose=False, mode="max"
+        monitor="val_auroc", patience=10, verbose=False, mode="max"
+    )
+
+    cp_callback = ModelCheckpoint(
+        './checkpoint/' + version, monitor='val_auroc', save_last=True, save_top_k=1, mode='max'
     )
@@ -359,3 +416,3 @@
         max_epochs=args.epochs,
-        callbacks=[early_stop_callback],
+        callbacks=[cp_callback]#, early_stop_callback],
     )
