--- thirdparty/BIOT/run_binary_supervised.py	2026-01-08 15:46:45.832706321 +0900
+++ finetune/BIOT/run_binary_supervised.py	2026-04-06 23:24:13.622032540 +0900
@@ -134,3 +134,3 @@
 
-    root = "/srv/local/data/TUH/tuh3/tuh_eeg_abnormal/v3.0.0/edf/processed"
+    root = "../../datasets/tuab/biot/"
 
@@ -350,2 +350,6 @@
 
+    cp_callback = ModelCheckpoint(
+        monitor='val_auroc', mode='max', save_top_k=2, save_last=True
+    )
+
     trainer = pl.Trainer(
@@ -359,3 +363,3 @@
         max_epochs=args.epochs,
-        callbacks=[early_stop_callback],
+        callbacks=[early_stop_callback, cp_callback],
     )
